src/gengowatcher/ui_textual.py

```
@@ -949,6 +1212,70 @@ def __init__(
+        try:
+            tabbed_content = self.query_one(TabbedContent)
+            active_tab_id = tabbed_content.active
+        except NoMatches:
+            # If TabbedContent can't be found, fall back to refreshing dashboard widgets.
+            active_tab_id = None
+
+        # Widgets that live on the dashboard tab.
+        dashboard_widgets = [
+            (MetricsRow, "refresh_metrics"),
+            (JobsPreview, "refresh_jobs"),
+            (JobsHourChart, "refresh_chart"),
```

Ruff (0.14.14)

Undefined name `JobsHourChart`

(F821)